### PR TITLE
Feature load improvement

### PR DESCRIPTION
**Pull request by michelcedric**
- Performance improvement, load all items in one call to database.
 - Change readibility : use lambda instead of for each

### CODE REVIEW

**Comment by michelcedric** [comment]
Hi @efleming18 
I took the liberty to change the load of some children (item). The repository was called for each item. I replace by one call.

**Comment by michelcedric** [comment]
@efleming18 Could you check this Pull Request?

File: src/Web/Services/BasketViewModelService.cs [hunk 1]
```diff
@@ -38,10 +39,13 @@ public async Task<BasketViewModel> GetOrCreateBasketForUser(string userName)
...
-            viewModel.Id = basket.Id;
-            viewModel.BuyerId = basket.BuyerId;
-            viewModel.Items = await GetBasketItems(basket.Items); ;
+            var viewModel = new BasketViewModel
+            {
+                Id = basket.Id,
+                BuyerId = basket.BuyerId,
+                Items = await GetBasketItems(basket.Items)
+            };
+            ;
```
Comment (efleming18) [inline src/Web/Services/BasketViewModelService.cs:48]: It looks like this was there before you, but could you remove this straggling semi-colon there?

Comment (michelcedric) [reply src/Web/Services/BasketViewModelService.cs]: Of course :)

**Comment by efleming18** [comment]
It also appears that the `RedirectsToLoginIfNotAuthenticated` is now not passing successfully. Could you take a look at that as well?  I verified that the test still passes on the `master` branch.

**Comment by michelcedric** [comment]
@efleming18 
Sorry, I make a mistake and not used the correct ID but it's fixed :)
Thnaks

**Comment by efleming18** [comment]
Looks good! Thanks, @michelcedric !

**Pull request merged**